File: pegasusX/apps/factory-app-android/app/src/main/java/com/pegasusx/factory/ui/navigation/FactorySection.kt
package com.pegasusx.factory.ui.navigation

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.ui.graphics.vector.ImageVector

enum class FactorySection(
    val route: String,
    val label: String,
    val icon: ImageVector,
) {
    DASHBOARD(FactoryRoutes.DASHBOARD, "Dashboard", Icons.Default.Dashboard),
    LOADING_BAY(FactoryRoutes.LOADING_BAY, "Loading Bay", Icons.Default.Inventory),
    TRANSFERS(FactoryRoutes.TRANSFERS, "Transfers", Icons.Default.SwapHoriz),
    FLEET(FactoryRoutes.FLEET, "Fleet", Icons.Default.LocalShipping),
    STAFF(FactoryRoutes.STAFF, "Staff", Icons.Default.People),
    LOCATION(FactoryRoutes.LOCATION_SETTINGS, "Location", Icons.Default.Place),
    SUPPLY_REQUESTS(FactoryRoutes.SUPPLY_REQUESTS, "Supply requests", Icons.Default.Sync),
    PAYLOAD_OVERRIDE(FactoryRoutes.PAYLOAD_OVERRIDE, "Payload override", Icons.Default.SwapHoriz),
    PAYLOAD_LOAD(FactoryRoutes.PAYLOAD_LOAD, "Payload / Load", Icons.Default.Inventory),
    MANIFESTS(FactoryRoutes.MANIFESTS, "Manifests", Icons.Default.Description),
    MANIFEST_EXCEPTIONS(FactoryRoutes.MANIFEST_EXCEPTIONS, "Gate exceptions", Icons.Default.Warning),
    INSIGHTS(FactoryRoutes.INSIGHTS, "Insights", Icons.Default.Insights),
    ANALYTICS(FactoryRoutes.ANALYTICS, "Analytics", Icons.Default.Analytics),
    NOTIFICATIONS(FactoryRoutes.NOTIFICATIONS, "Notifications", Icons.Default.Notifications),
    MORE(FactoryRoutes.MORE, "More", Icons.Default.Apps),
    ;

    companion object {
        val compactTabs: List<FactorySection> = listOf(DASHBOARD, LOADING_BAY, TRANSFERS, MORE)

        val primarySections: List<FactorySection> = listOf(
            DASHBOARD, LOADING_BAY, TRANSFERS, FLEET, STAFF, LOCATION,
        )

        val operationsSections: List<FactorySection> = listOf(
            SUPPLY_REQUESTS, PAYLOAD_LOAD, PAYLOAD_OVERRIDE, MANIFESTS, MANIFEST_EXCEPTIONS,
        )

        val intelligenceSections: List<FactorySection> = listOf(
            INSIGHTS, ANALYTICS, NOTIFICATIONS,
        )

        /** Tablet rail — mirrors iOS sidebar + factory-portal nav. */
        val drawerSections: List<FactorySection> =
            primarySections + operationsSections + intelligenceSections

        /** Phone More hub: everything not on the bottom tab bar (except More itself). */
        val moreHubSections: List<FactorySection> =
            primarySections.filter { it !in compactTabs.filter { tab -> tab != MORE } } +
                operationsSections +
                intelligenceSections

        fun fromRoute(route: String?): FactorySection? {
            if (route.isNullOrBlank()) return null
            val exact = entries.firstOrNull { it.route == route }
            if (exact != null) return exact
            return when {
                route == FactoryRoutes.TRANSFER_CREATE -> TRANSFERS
                route.startsWith("transfers/") -> TRANSFERS
                route.startsWith("manifests/") -> MANIFESTS
                route.startsWith("staff/") -> STAFF
                else -> {
                    val base = route.substringBefore("/").substringBefore("?")
                    entries.firstOrNull { it.route == base }
                }
            }
        }
    }
}

File: pegasusX/apps/factory-app-ios/FactoryApp/Layout/FactorySection.swift
import SwiftUI

enum FactorySection: String, CaseIterable, Identifiable {
    case dashboard = "Dashboard"
    case loadingBay = "Loading Bay"
    case transfers = "Transfers"
    case fleet = "Fleet"
    case staff = "Staff"
    case location = "Location"
    case supplyRequests = "Supply requests"
    case payloadOverride = "Override"
    case manifests = "Manifests"
    case manifestExceptions = "Exceptions"
    case insights = "Insights"
    case analytics = "Analytics"
    case notifications = "Notifications"

    var id: String { rawValue }

    var icon: String {
        switch self {
        case .dashboard: "square.grid.2x2"
        case .loadingBay: "shippingbox"
        case .transfers: "arrow.left.arrow.right"
        case .fleet: "truck.box"
        case .staff: "person.2"
        case .location: "mappin.and.ellipse"
        case .supplyRequests: "checklist"
        case .payloadOverride: "arrow.left.arrow.right.square"
        case .manifests: "doc.text"
        case .manifestExceptions: "exclamationmark.triangle"
        case .insights: "chart.bar.xaxis"
        case .analytics: "chart.xyaxis.line"
        case .notifications: "bell"
        }
    }

    static var compactTabs: [FactorySection] { [.dashboard, .loadingBay, .transfers] }

    static var primarySections: [FactorySection] {
        [.dashboard, .loadingBay, .transfers, .fleet, .staff, .location]
    }

    static var operationsSections: [FactorySection] {
        [.supplyRequests, .payloadOverride, .manifests, .manifestExceptions]
    }

    static var intelligenceSections: [FactorySection] {
        [.insights, .analytics, .notifications]
    }
}

enum FactoryCompactTab: Hashable {
    case dashboard
    case loadingBay
    case transfers
    case more
}
